cli/service/service.go

```diff
@@ -135,17 +132,35 @@ func Log(serviceName string, tailing bool) {
 					cliutils.Fatal(cliutils.CLI_INPUT_ERROR, msgPrinter.Sprintf("Deployment unmarshalling error: %v", err))
 				}
 
-				for _, service := range deployment.Services {
-					var err error
-					if nonDefaultLogDriverUsed, err = cliutils.ChekServiceLogPossibility(service.LogDriver); err != nil {
-						cliutils.Fatal(cliutils.CLI_GENERAL_ERROR, msgPrinter.Sprintf("Service logs are unavailable: %v", err))
+				if len(deployment.Services) > 1 && containerName == "" {
+					cliutils.Fatal(cliutils.CLI_INPUT_ERROR, msgPrinter.Sprintf("Service definition %v consists of more than one container. Please specify the container name using the -c flag.", serviceName))
+				}
+				for deployedServiceName, service := range deployment.Services {
+					if containerName == deployedServiceName || (containerName == "" && len(deployment.Services) == 1) {
+						var err error
+						if nonDefaultLogDriverUsed, err = cliutils.ChekServiceLogPossibility(service.LogDriver); err != nil {
+							cliutils.Fatal(cliutils.CLI_GENERAL_ERROR, msgPrinter.Sprintf("Service logs are unavailable: %v", err))
+						}
+						containerFound = true
+						break
 					}
-					break
 				}
 			}
 			break
 		}
 	}
+	if !containerFound && containerName != "" {
+		cliutils.Fatal(cliutils.CLI_INPUT_ERROR, msgPrinter.Sprintf("Container %v is not running as part of service %v.", containerName, serviceName))
+	} else if !containerFound {
+		cliutils.Fatal(cliutils.CLI_INPUT_ERROR, msgPrinter.Sprintf("Could not find service %v running on the node.", containerName, serviceName))
+	} else {
+		msgPrinter.Printf("Displaying log messages of container %v for service %v with service id %v.", containerName, name, instanceId)
```

Comment:
This message comes out with a blank container name when the service's container is found but no containerName was provided as input. I think you will need to set containerName around line 144.